tests/sparseml/pytorch/sparsification/distillation/test_per_layer_distillation.py

```diff
@@ -0,0 +1,202 @@
+            test_steps_per_epoch,
+            distillation_teacher=model_lambda(),
+        )
+
+    def test_scheduled_update(
+        self,
+        modifier_lambda: Callable[[], ScheduledModifier],
+        model_lambda: Callable[[], Module],
+        optim_lambda: Callable[[Module], Optimizer],
+        test_epoch: float,  # noqa: F811
+        test_steps_per_epoch: int,  # noqa: F811
+    ):
+        super().test_scheduled_update(
+            modifier_lambda,
+            model_lambda,
+            optim_lambda,
+            test_epoch,
+            test_steps_per_epoch,
+            distillation_teacher=model_lambda(),
+        )
+
+    def test_lifecycle(
+        self,
+        modifier_lambda,
+        model_lambda,
+        optim_lambda,
+        test_steps_per_epoch,  # noqa: F811
+    ):
+        modifier = modifier_lambda()
+        model = model_lambda()
+        optimizer = optim_lambda(model)
+
+        self.initialize_helper(modifier, model, distillation_teacher=model_lambda())
+
+        for epoch in range(int(modifier.start_epoch)):
+            assert not modifier.update_ready(epoch, test_steps_per_epoch)
+
+        assert modifier.update_ready(modifier.start_epoch, test_steps_per_epoch)
+        modifier.scheduled_update(
+            model, optimizer, modifier.start_epoch, test_steps_per_epoch
+        )
+
+        if modifier.end_epoch > modifier.start_epoch:
+            assert not modifier.update_ready(
+                (modifier.start_epoch + modifier.end_epoch) / 2, test_steps_per_epoch
+            )
+            assert modifier.update_ready(modifier.end_epoch, test_steps_per_epoch)
+
+    def test_loss_update(
+        self,
+        modifier_lambda: Callable[[], Modifier],
+        model_lambda: Callable[[], Module],
+        optim_lambda: Callable[[Module], Optimizer],
+        test_epoch: float,  # noqa: F811
+        test_steps_per_epoch: int,  # noqa: F811
+        test_loss: Tensor,  # noqa: F811
+    ):
+        modifier = modifier_lambda()
+        model = model_lambda()
+        optimizer = optim_lambda(model)
+
+        self.initialize_helper(modifier, model, distillation_teacher=model_lambda())
+
+        # test distillation has been applied
+        # fake forward pass
+        student_inputs = _get_fake_batch(model_lambda)
+        student_outputs = model(student_inputs)
+        fake_loss = student_outputs.mean()
+        updated_loss = modifier.loss_update(
+            fake_loss,
+            model,
+            optimizer,
+            modifier.start_epoch,
+            test_steps_per_epoch,
+            student_outputs,
+            student_inputs,
+        )
+
+        assert isinstance(updated_loss, torch.Tensor)
+        assert updated_loss.shape == fake_loss.shape
+        assert fake_loss.item() != updated_loss.item()
+
+
+@pytest.mark.skipif(
+    os.getenv("NM_ML_SKIP_PYTORCH_TESTS", False),
+    reason="Skipping pytorch tests",
+)
+def test_distillation_modifier_yaml():
```

Comment:
I think for this test we just need to:
1. Change the name of the class being used to the per layer one
2. Add all the possible fields